perfkitbenchmarker/linux_benchmarks/tensorflow_benchmark.py

```diff
@@ -0,0 +1,257 @@
+                    'Name of dataset: imagenet or flowers.')
+flags.DEFINE_integer('batch_size', 64, 'batch size per compute device')
+flags.DEFINE_string('variable_update', 'parameter_server',
+                    '''The method for managing variables: parameter_server,
+                    replicated, distributed_replicated, independent''')
+flags.DEFINE_string('local_parameter_device', 'gpu',
+                    '''Device to use as parameter server: cpu or gpu. For
+                    distributed training, it can affect where caching of
+                    variables happens.''')
+flags.DEFINE_boolean('use_nccl', True,
+                     'Whether to use nccl all-reduce primitives where possible')
+flags.DEFINE_boolean('distortions', True,
+                     '''Enable/disable distortions during image preprocessing.
+                     These include bbox and color distortions.''')
+
+
+class TFParseOutputException(Exception):
+  pass
+
+
+def GetConfig(user_config):
+  """Load and return benchmark config.
+
+  Args:
+    user_config: user supplied configuration (flags and config file)
+
+  Returns:
+    loaded benchmark configuration
+  """
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def CheckPrerequisites(_):
+  """Verify that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  cuda_toolkit_8.CheckPrerequisites()
+
+
+def _LocalDataPath(local_file):
+  """Return local data path for given file.
+
+  Args:
+    local_file: name of local file to create full data path for
+
+  Returns:
+    path of local_file, in the data directory
+  """
+  return data.ResourcePath(local_file)
+
+
+def _UpdateBenchmarkSpecWithFlags(benchmark_spec):
+  """Update the benchmark_spec with supplied command line flags.
+
+  Args:
+    benchmark_spec: benchmark specification to update
+  """
+  benchmark_spec.model = FLAGS.model
+  benchmark_spec.data_name = FLAGS.data_name
+  benchmark_spec.batch_size = FLAGS.batch_size
+  benchmark_spec.variable_update = FLAGS.variable_update
+  benchmark_spec.local_parameter_device = FLAGS.local_parameter_device
+  benchmark_spec.use_nccl = FLAGS.use_nccl
+  benchmark_spec.distortions = FLAGS.distortions
+
+
+def Prepare(benchmark_spec):
+  """Install and set up TensorFlow on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification
+  """
+  _UpdateBenchmarkSpecWithFlags(benchmark_spec)
+  vms = benchmark_spec.vms
+  master_vm = vms[0]
+  logging.info('Installing CUDA Toolkit 8.0 on %s', master_vm)
+  master_vm.Install('cuda_toolkit_8')
+  benchmark_spec.num_gpus = cuda_toolkit_8.QueryNumberOfGpus(master_vm)
+  master_vm.Install('cudnn')
+  master_vm.Install('tensorflow')
+
+
+def _CreateMetadataDict(benchmark_spec):
+  """Create metadata dict to be used in run results.
+
+  Args:
+    benchmark_spec: benchmark spec
+
+  Returns:
+    metadata dict
+  """
+  metadata = dict()
```

Comment:
We want the clock settings in the metadata. Add `metadata.update(cuda_toolkit_8.GetMetadataFromFlags())`